### PR TITLE
Implement shutdown on ping timeout

### PR DESCRIPTION
**Pull request by m-novikov**

### CODE REVIEW

File: tiktorch/server/base.py [hunk 1]
```diff
@@ -23,9 +21,14 @@ def main():
     parsey.add_argument("--debug", action="store_true")
     parsey.add_argument("--dummy", action="store_true")
     parsey.add_argument("--connection-file", help="where to write connection parameters file")
-    parsey.add_argument("--kill-timeout", type=int, default=KILL_TIMEOUT)
+    parsey.add_argument(
+        "--kill-timeout",
+        type=float,
+        default=0.0,
```
Comment (k-dominik) [inline tiktorch/server/base.py:27]: I'm a bit weary of the special meaning of `0.0` here. It's not clear from the help string, or the docstring of the function that `0.0` is a special case and that there is no timeout at all (never killing the server), whereas `0.0000001` would work (which in practice is no different than 0.0, I assume). In order to find this out one has to check the code or the test. I'd say that's a bit much for a command line interface.

File: tiktorch/server/grpc/__init__.py [hunk 2]
```diff
@@ -16,7 +16,14 @@
 from .inference_servicer import InferenceServicer
 
 
-def serve(host, port, *, connection_file_path: Optional[str] = None):
+def serve(host, port, *, connection_file_path: Optional[str] = None, kill_timeout: Optional[int] = None):
```
Comment (k-dominik) [inline tiktorch/server/grpc/__init__.py:19]:
```suggestion
def serve(host, port, *, connection_file_path: Optional[str] = None, kill_timeout: Optional[float] = None):
```

**Review by k-dominik — COMMENTED** [review]
Good to see there will not be any dangling tiktorches on user's local machines with this. Left a few minor comments.

File: tiktorch/server/base.py (hunk 1 above)
Comment (m-novikov) [reply tiktorch/server/base.py]: Now use `None` instead of `0.0`

**Review by k-dominik — APPROVED** [review]
lgtm

**Pull request merged**